crates/services/gas_price_service/src/fuel_gas_price_updater/fuel_core_storage_adapter.rs

```diff
@@ -0,0 +1,183 @@
+use crate::fuel_gas_price_updater::{
+    fuel_core_storage_adapter::database::{
+        GasPriceColumn,
+        GasPriceMetadata,
+    },
+    BlockInfo,
+    Error,
+    Error as GasPriceError,
+    L2BlockSource,
+    MetadataStorage,
+    Result,
+    Result as GasPriceResult,
+    UpdaterMetadata,
+};
+use anyhow::anyhow;
+use fuel_core_services::stream::BoxStream;
+use fuel_core_storage::{
+    kv_store::KeyValueInspect,
+    structured_storage::StructuredStorage,
+    transactional::{
+        Modifiable,
+        WriteTransaction,
+    },
+    StorageAsMut,
+    StorageAsRef,
+};
+use fuel_core_types::fuel_types::BlockHeight;
+
+use fuel_core_types::{
+    blockchain::{
+        block::Block,
+        header::ConsensusParametersVersion,
+    },
+    fuel_tx::{
+        field::{
+            MintAmount,
+            MintGasPrice,
+        },
+        Transaction,
+    },
+    services::block_importer::SharedImportResult,
+};
+use std::cmp::min;
+use tokio_stream::StreamExt;
+
+#[cfg(test)]
+mod metadata_tests;
+
+#[cfg(test)]
+mod l2_source_tests;
+
+pub mod database;
```

Comment:
```suggestion
pub mod storage;
```

It would be nice to use "storage" naming since database is more concrete=)